test/hotspot/jtreg/compiler/vectorapi/VectorMaskCastTest.java

```diff
@@ -0,0 +1,462 @@
+/*
+ * Copyright (c) 2021, Arm Limited. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.vectorapi;
+
+import java.util.Random;
+
+import jdk.incubator.vector.ByteVector;
+import jdk.incubator.vector.DoubleVector;
+import jdk.incubator.vector.FloatVector;
+import jdk.incubator.vector.IntVector;
+import jdk.incubator.vector.LongVector;
+import jdk.incubator.vector.ShortVector;
+import jdk.incubator.vector.VectorMask;
+import jdk.incubator.vector.VectorSpecies;
+import jdk.test.lib.Utils;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+/**
+ * @test
+ * @bug 8273264
+ * @library /test/lib
+ * @summary AArch64: [vector] Add missing rules for VectorMaskCast
+ * @modules jdk.incubator.vector
+ *
+ * @run testng/othervm -XX:-TieredCompilation compiler.vectorapi.VectorMaskCastTest
+ */
+
+
+public class VectorMaskCastTest{
```

Comment:
Do we need to add such a test? We already have conversions tests, see sub-classes of `AbstractVectorConversionTest`.

Reply:
I tested with those conversion tests before, but the method could not be intrinsified. I checked generated code and the new test cases could work.

Reply:
Do you have any indicates as to why? Is it because the invocation count is too low?

Reply:
The reason maybe the same as https://bugs.openjdk.java.net/browse/JDK-8259610, but I didn't pay too much time in the details.

Reply:
Ok. I don't wanna block this PR. Could you add some comment to the test providing some rational as to why it is currently needed, making reference to the vector conversions tests, and that we need eventually resolve such issues to avoid the duplication?

Reply:
Yes, it's been added in the new commit. Thanks for your kindly review.